fuse/api.go

```diff
@@ -292,6 +292,8 @@ type RawFileSystem interface {
 
 	StatFs(cancel <-chan struct{}, input *InHeader, out *StatfsOut) (code Status)
 
+	Ioctl(cancel <-chan struct{}, in *IoctlIn, data []byte) Status
```

Comment:
Why not use IoctlOut?

Reply:
It seems we are always in restricted ioctl, and the IoctlOut makes no sense to the user.  
https://elixir.bootlin.com/linux/v5.19.2/source/fs/fuse/ioctl.c#L152

Reply:
It's better to follow the FUSE protocol, don't over optimize it for current task, that make us easier to upstream this patch.

Reply:
`Ioctl(cancel <-chan struct{}, in *IoctlIn, out *IoctlOut, data []byte) Status`  like this?

Reply:
> For security and cleanliness considerations, ioctl implementation has
restricted mode where the kernel determines data transfer directions
and sizes using the _IOC_*() macros on the ioctl command.  In this
mode, retry is not allowed.
>
> For all FUSE servers, restricted mode is enforced.  Unrestricted ioctl
will be used by CUSE.

FYI. https://sourceforge.net/p/fuse/mailman/message/20852313/

Reply:
I think we can follow libfuse: https://github.com/libfuse/libfuse/blob/master/lib/fuse_lowlevel.c#L943

`data [][]byte` can be part of `IoctlIn` and `IoctlOut`, it will be nil in most of the cases.

Reply:
`fuse_reply_ioctl_retry` is expected to be used in cuse. And we can't add fields in IoctlIn and IoctlOut because that will change the size of the struct which used here https://github.com/juicedata/go-fuse/blob/183cb227b35372c7ea50bc752d8159df5fdad5dd/fuse/opcode.go#L517

Reply:
I see, can we use the following one?
```
Ioctl(cancel <-chan struct{}, in *IoctlIn,  out *IoctlOut, bufIn []byte, bufOut []byte) Status
```

Reply:
Ok. But in restricted mode, only one directon exists (read or write).  it seems a little redundancy

Reply:
done